### PR TITLE
HTML: use Wasm to get a SharedArrayBuffer instance

### PR DESCRIPTION
**Pull request by annevk**
And verify that the SharedArrayBuffer constructor isn't exposed without COOP+COEP.

For https://github.com/web-platform-tests/wpt/issues/22358.

### CODE REVIEW

File: html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/no-coop-coep.https.any.js [hunk 1]
```diff
@@ -1,18 +1,29 @@
+// META: global=window,worker
+
+test(() => {
+  // See https://github.com/whatwg/html/issues/5380 for why not `new SharedArrayBuffer()`
+  assert_equals(globalThis.SharedArrayBuffer, undefined);
+  assert_false("SharedArrayBuffer" in globalThis);
+}, "SharedArrayBuffer constructor does not exist without COOP+COEP");
```
Comment (annevk) [inline html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/no-coop-coep.https.any.js]: @syg @domenic any other angles to test?

Comment (domenic) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/no-coop-coep.https.any.js]: Nah that should do it.

Comment (syg) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/no-coop-coep.https.any.js]: lgtm

**Review by syg — APPROVED** [review]
SAB changes themselves lgtm, with caveat I don't really know how WPTs work.

**Review by jgraham — APPROVED** [review]

**Pull request merged**